src/ResourceManager/Websites/Commands.Websites/Cmdlets/BackupRestore/RestoreAzureWebAppSnapshot.cs

```diff
@@ -0,0 +1,77 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using Microsoft.Azure.Management.WebSites.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.BackupRestore
+{
+    /// <summary>
+    ///     Restores an Azure Web App snapshot
+    /// </summary>
+    [Cmdlet(VerbsData.Restore, "AzureRmWebAppSnapshot")]
+    public class RestoreAzureWebAppSnapshot : WebAppOptionalSlotBaseCmdlet
+    {
+        [Parameter(Position = 3, Mandatory = true, HelpMessage = "The timestamp of the snapshot.", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string SnapshotTime;
+
+        [Parameter(Mandatory = false, HelpMessage = "Recover the web app's configuration in addition to files.", ValueFromPipelineByPropertyName = true)]
+        public SwitchParameter RecoverConfiguration { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The app that the snapshot contents will be restored to. Must be a slot of the original app. If unspecified, the original app is overwritten.", ValueFromPipelineByPropertyName = true)]
+        public Site TargetApp { get; set; }
```

Comment:
@Nking92 instead of having the user passing in an entire `Site` object (which requires a separate cmdlet call to get), would we be able to change this to `TargetSiteName` and have the user provide a `string` instead?

Reply:
We will need the entire ARM resource ID since the other site could be in any resource group. I thought using the Site object would be the easiest way to use the API. If reducing the amount of network traffic / roundtrips is desired, I can add a separate parameter set that uses a string resource ID.